src/sc_client/session.py

```diff
@@ -23,35 +23,53 @@
     MAX_PAYLOAD_SIZE,
     SERVER_ANSWER_CHECK_TIME,
     SERVER_ESTABLISH_CONNECTION_TIME,
-    SERVER_RECONNECTION_TIME,
+    SERVER_RECONNECT_RETRIES,
+    SERVER_RECONNECT_RETRY_DELAY,
 )
 from sc_client.models import Response, ScAddr, ScEvent
 
 logger = logging.getLogger(__name__)
 
 
+def default_reconnect_handler() -> None:
+    establish_connection(_ScClientSession.ws_app.url)
+
+
 class _ScClientSession:
+    is_open = False
     lock_instance = threading.Lock()
     responses_dict = {}
     events_dict = {}
     command_id = 0
     executor = Executor()
     ws_app: websocket.WebSocketApp | None = None
-    do_reconnect = False
-    reconnecion_time = SERVER_RECONNECTION_TIME
+    error_handler = None
+    reconnect_callback = default_reconnect_handler
+    post_reconnect_callback = None
+    reconnect_retries: int = SERVER_RECONNECT_RETRIES
+    reconnect_retry_delay: float = SERVER_RECONNECT_RETRY_DELAY
+    last_healthcheck_answer: str = None
 
     @classmethod
     def clear(cls):
+        cls.is_open = False
         cls.responses_dict = {}
         cls.events_dict = {}
         cls.command_id = 0
         cls.ws_app = None
+        cls.error_handler = None
+        cls.reconnect_callback = default_reconnect_handler
+        cls.post_reconnect_callback = None
+        cls.reconnect_retries = SERVER_RECONNECT_RETRIES
+        cls.reconnect_retry_delay = SERVER_RECONNECT_RETRY_DELAY
 
 
 def _on_message(_, response: str) -> None:
     logger.debug(f"Receive: {str(response)[:LOGGING_MAX_SIZE]}")
     response = json.loads(response, object_hook=Response)
-    if response.get(common.EVENT):
+    if isinstance(response, str):
+        _ScClientSession.last_healthcheck_answer = response
```

Comment:
What's the pragmatic side of this decision? If this variable is only used for healthcheck requests, can we just make a wrapper method for this (or reuse an existing method, if there is one already) rather than writing a variable every time we get a websocket message? This looks excessive to me, but maybe I didn't get your idea on this one